tsdb/index/postings.go

```diff
@@ -49,46 +50,63 @@ var ensureOrderBatchPool = sync.Pool{
 	},
 }
 
+const shardCount = 256
+
+type memPostingMap struct {
+	mtx    *sync.RWMutex
```

Comment:
Does this have to be a pointer?

Reply:
Technically no, but it allows us to have this block:

```
        for _, shard := range p.m {
		shard.mtx.RLock()
		for n, e := range shard.series {
			for v := range e {
				keys = append(keys, labels.Label{Name: n, Value: v})
			}
		}
		shard.mtx.RUnlock()
	}
```

If it wasn't a pointer we would have to access the shard struct via index (`p.m[i].mtx.RLock()`) every time, which I think is less readable and more likely to introduce a bug someday.

Reply:
How about
```
for i := range p.m {
  shard := &p.m[i]
```

Reply:
Updated to just use index directly, it's only a few places